csi/util.go

```diff
@@ -239,6 +239,26 @@ func parseJSONRecurringJobs(jsonRecurringJobs string) ([]longhornclient.Recurrin
 	return recurringJobs, nil
 }
 
+func syncMountPointDirectory(targetPath string) error {
+	d, err := os.Open(targetPath)
+	if err != nil {
+		return err
+	}
+	defer d.Close()
+
+	if _, err := d.Readdirnames(-1); err != nil {
+		return err
+	}
+
+	// it would not always return `Input/Output Error` or `read-only file system` errors if we only use Sync() without the previous I/O operation
+	// and it needs more investigations why only Sync() would not return an error
+	if err := d.Sync(); err != nil {
+		return err
+	}
+
+	return d.Close()
```

Comment:
Isn't this already handled by the [defer d.Close()](https://github.com/longhorn/longhorn-manager/blob/aae8aac5e8d1321670b461ede38436f470418532/csi/util.go#L247)?

Reply:
Yes, you are right. Modified. Thanks